torch/utils/data/dataset.py

```diff
@@ -35,3 +35,51 @@ def __getitem__(self, index):
 
     def __len__(self):
         return self.data_tensor.size(0)
+
+
+class ConcatDataset(Dataset):
+    """
+    Dataset to concatenate multiple datasets.
+    Purpose: useful to assemble different existing datasets, possibly
+    large-scale datasets as the concatenation operation is done in an
+    on-the-fly manner.
+    Args:
+        datasets (iterable): List of datasets to be concatenated
+    """
+
+    @staticmethod
+    def cumsum(sequence):
+        result = []
+        sum = 0
+        for s in sequence:
+            l = len(s)
+            cumsum.append(l + sum)
+            sum += l
+        return result
```

Comment:
This function is not necessary, replace it by `sum([len(sequence) for sequence for sequences])`.